contracts/HubController.sol

```diff
@@ -0,0 +1,116 @@
+
+contract HubController is Named, Versioned, ContractStatus, Ownable {
+    string private constant _NAME = "HubController";
+    string private constant _VERSION = "1.0.0";
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor(address hubAddress) ContractStatus(hubAddress) {}
+
+    // @dev Only transactions by HubController owner or one of the owners of the MultiSig Wallet
+    modifier onlyOwnerOrMultiSigOwner() {
+        _checkOwnerOrMultiSigOwner();
+        _;
+    }
+
+    function name() external pure virtual override returns (string memory) {
+        return _NAME;
+    }
+
+    function version() external pure virtual override returns (string memory) {
+        return _VERSION;
+    }
+
+    function forwardCall(address target, bytes calldata data) external onlyOwnerOrMultiSigOwner returns (bytes memory) {
+        require(hub.isContract(target), "Target contract isn't in the Hub");
+
+        (bool success, bytes memory result) = target.call{value: 0}(data);
+
+        if (!success) {
+            assembly {
+                let ptr := mload(0x40)
+                let size := returndatasize()
+                returndatacopy(ptr, 0, size)
+                revert(ptr, size)
+            }
+        }
+
+        return result;
+    }
+
+    function setAndReinitializeContracts(
+        GeneralStructs.Contract[] calldata newContracts,
+        GeneralStructs.Contract[] calldata newAssetStorageContracts,
+        address[] calldata contractsToReinitialize
+    ) external onlyOwnerOrMultiSigOwner {
+        for (uint i; i < newContracts.length; ) {
+            hub.setContractAddress(newContracts[i].name, newContracts[i].addr);
+            unchecked {
+                i++;
+            }
+        }
+
+        for (uint i; i < newAssetStorageContracts.length; ) {
+            hub.setAssetStorageAddress(newAssetStorageContracts[i].name, newAssetStorageContracts[i].addr);
+            unchecked {
+                i++;
+            }
+        }
+
+        for (uint i; i < contractsToReinitialize.length; ) {
+            Initializable(contractsToReinitialize[i]).initialize();
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    function setContractAddress(
+        string calldata contractName,
+        address newContractAddress
+    ) external onlyOwnerOrMultiSigOwner {
+        hub.setContractAddress(contractName, newContractAddress);
+    }
+
+    function setAssetStorageAddress(
+        string calldata assetStorageName,
+        address assetStorageAddress
+    ) external onlyOwnerOrMultiSigOwner {
+        hub.setAssetStorageAddress(assetStorageName, assetStorageAddress);
+    }
+
+    function renounceHubOwnership() external onlyOwner {
+        hub.renounceOwnership();
+    }
+
+    function transferHubOwnership(address newOwner) external onlyOwner {
+        hub.transferOwnership(newOwner);
+    }
+
+    function _isMultiSigOwner() internal view returns (bool) {
```

Comment:
Can we add some tests to cover this function, and change of multisig owners, maybe intergation tests